lib/utils/makeValidator.js

```diff
@@ -18,6 +21,72 @@ const makeLinks = (error, makerFunc) => {
   return [];
 };
 
+const removeFirstAndLastChar = s => s.slice(1, -1);
+// always return a string
+const makePath = (path, newSegment) =>
+  (path ? [path, newSegment].join('.') : newSegment) || '';
+
+const processBaseError = (err, path) => {
+  const completePath = makePath(path, err.property).replace(
+    /\.instance\./g,
+    '.'
+  );
+
+  err.property = completePath;
+  return err;
+};
+
+const cleanError = (validationError, path, validator) => {
+  if (ambiguousTypes.includes(validationError.name)) {
+    // TODO: make this smarter?
+    // if it's triggerSchema, check for "type"
+    // if it's request/function, check for "source" or "require". if neither are present, validate against request
+
+    // otherwise, maybe raise regular and drop a note to file an issue so we can consider that case?
+
+    let nextSchema;
+    // if (
+    //   isEqual(validationError.argument, [
+    //     '</BasicPollingOperationSchema>',
+    //     '</BasicHookOperationSchema>'
+    //   ])
+    // ) {
+    //   // we can take advantage of type being optional on polling but required on hooks
+    //   nextSchema =
+    //     validationError.argument[
+    //       validationError.instance.type === 'hook' ? 1 : 0
+    //     ];
+    // } else {
+    nextSchema = validationError.argument[0];
+    // }
+
+    // we have some schemas with anonymous subschemas (such as FieldChoices)
+    if (nextSchema.includes('[subschema')) {
+      const namedSchema = validator.schemas[validationError.schema];
+
+      if (!namedSchema) {
+        return processBaseError(validationError, path);
```

Comment:
If I do this for a `perform`:

```
perform: {
  url: 'https://example.com',
  body: 123
}
```

I see this error:

```
┌────────────────────────────────────────────────────────────────────────────────────────────────────────────────┐
│ = 1 =                                                                                                          │
│     Property │ App.triggers.movie.operation.perform.body                                                       │
│     Message  │ is not exactly one from [subschema 0],[subschema 1],[subschema 2],[subschema 3]                 │
│     Links    │ https://github.com/zapier/zapier-platform-schema/blob/v7.6.1/docs/build/schema.md#[subschema 0] │
│              │ https://github.com/zapier/zapier-platform-schema/blob/v7.6.1/docs/build/schema.md#[subschema 1] │
│              │ https://github.com/zapier/zapier-platform-schema/blob/v7.6.1/docs/build/schema.md#[subschema 2] │
│              │ https://github.com/zapier/zapier-platform-schema/blob/v7.6.1/docs/build/schema.md#[subschema 3] │
└──────────────┴─────────────────────────────────────────────────────────────────────────────────────────────────┘
```

Is it possible to replace `[subschema x]` with a more readable text?

Reply:
maybe! i noticed the same thing. I think it's a problem of going one level too far down. I'll jump back in, but my guess is that the error structure is something like

```js
{
  argument: ['subschema 0', 'subschema 1', ...],
  schema: [{type: 'number'}, {type: 'object'}... ]
  ...
}
```

so we could pull those types out. I agree with the above that we should be careful doing this too coupled to the structure of the schema itself, but some of these edge cases are pretty wild.